### PR TITLE
Added getDialogueByDialogueName function

### PR DESCRIPTION
**Pull request by PineappleBrain**
Convenient! So you can composite a string and use this function.

### CODE REVIEW

**Comment by m0dE** [comment]
Can you explain the usecase for this? Ideally, you should be able to reference Dialogue by global variable / entity variable

**Comment by PineappleBrain** [comment]
It can benefit games that widely use dialogue.
---
UC1
A game that with quest or challenge. Lets say it have 100 quest, and each is unique.
Then the create can use `\`dialogue${num}\`` to get the dialogue.
---
UC2
A game with story
Then create can play dialogues page by page `\`dialogue${num++}\`` instead of clicking many times

Altho dialogue can use global variable but it is not very convenient.
JSON array is also having poor performance and junior creators dont know how to use
Referencing a dialogue is good idea. But not referencing many dialogues as creators still need to update follow up dialogue or similar setting with glob one by one which is the having same time cost.

So It seems not very friendly to store the hashed key one by one for dialogues.

Similar idea I want to implement in state of entity too. Im sure some creators have need on it

File: src/gameClasses/components/script/VariableComponent.js [hunk 1]
```diff
@@ -895,6 +895,21 @@ var VariableComponent = TaroEntity.extend({
 					returnValue = taro.$(id);
 					break;
 
+				case 'getDialogueByDialogueName': // return the first found dialogue with the given string
+					var name = self.getValue(text.name, vars);
+					var dialogues = taro.game.data.dialogues;
+					for (const id in dialogues) {
+						console.log(dialogues[id].name, name);
```
Comment (420kyle69) [inline src/gameClasses/components/script/VariableComponent.js:902]: You should remove console.log().

**Review by 420kyle69 — CHANGES_REQUESTED** [review]

**Comment by PineappleBrain** [comment]
Use dynamically generated dialogue content would be a better option instead

**Pull request closed without merging**